src/compiler.ts

```diff
@@ -10763,7 +10763,8 @@ export class Compiler extends DiagnosticEmitter {
       case TypeKind.EXTERNREF: {
         // TODO: non-null object might still be considered falseish
         // i.e. a ref to Boolean(false), Number(0), String("") etc.
-        return module.unary(UnaryOp.EqzI32, module.ref_is_null(expr));
+        // TODO: return module.unary(UnaryOp.EqzI32, module.ref_is_null(expr));
+        assert(false, 'Truthy checks for externref are not yet supported');
```

Comment:
Looks like these should be diagnostics, i.e. `DiagnosticCode.Not_implemented_0` with message `"ref.is_null"`. Unfortunately `makeIsTrueish` and others don't take the AST node as an argument yet (typically as `reportNode: Node`), which would have to be added in order to obtain the diagnostic range.

Reply:
>Unfortunately makeIsTrueish and others don't take the AST node as an argument yet 
Yep that's why I punted on it. How should I proceed?

Reply:
ohh I think you mean that I should pass in the appropriate node as another argument of `reportNode: Node` just for reporting purposes. That's hairy but makes sense. Will do. Lmk if that's wrong.

Reply:
Suggesting to add a `reportNode: Node` parameter to the affected functions, and provide it where these are called. The diagnostics would then look like:

```ts
this.error(
  DiagnosticCode.Not_implemented_0,
  reportNode.range,
  "ref.is_null" // respectively "ref.null", i.e. the exact feature missing
);
return module.unreachable();
```

replacing the `assert`s.

Reply:
Done. please review.